File: Easy/Array Pair Sum Divisibility Problem.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution
{
  public:
    bool canPair(vector<int> nums, int k)
    {
        // Code here.
        int n = nums.size();
        map<int, int> mappa;
        
        for (int i=0; i<n; i++)
        {
            mappa[nums[i] % k]++;
        }
        
        for (auto it: mappa)
        {
            if (it.first == 0 && it.second %2 == 0) continue;
            
            if (mappa[abs(k-it.first)] == it.second) continue;
            
            return false;
        }
        
        return true;
    }
};

//{ Driver Code Starts.
int main() {
    int tc;
    cin >> tc;
    while (tc--) {
        int n, k;
        cin >> n >> k;
        vector<int> nums(n);
        for (int i = 0; i < nums.size(); i++) cin >> nums[i];
        Solution ob;
        bool ans = ob.canPair(nums, k);
        if (ans)
            cout << "True\n";
        else
            cout << "False\n";
    }
    return 0;
}
// } Driver Code Ends

File: Medium/Array Pair Sum Divisibility Problem.py
#User function Template for python3

class Solution:
	def canPair(self, nuns, k):
		# Code here
	    n = len(nums)
	    
	    if n % 2 != 0:
            return False
        
        for i in range(0, n, 2):
            if (nums[i] + nums[i + 1]) % k != 0:
                return False
                
        return True
	    


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
	T=int(input())
	for i in range(T):
		n, k = input().split()
		n = int(n)
		k = int(k)
		nums = list(map(int, input().split()))
		ob = Solution()
		ans = ob.canPair(nums, k)
		if(ans):
			print("True")
		else:
			print("False")
# } Driver Code Ends